### ISSUE TITLE
Contour is not filling at larger scales

### ISSUE THREAD

**Issue by skorasaurus** [ccrook/QGIS-Contour-Plugin#16]
Hi, 

I'm running qgis 2.14.0 on ubuntu; python 2.7.6; 

I'm using the contour plugin on a 26k point geojson file. 

After running the contour plugin on the file, the contours all display properly at a small scales (1:70,000).
However, when I zoom into a larger scale, many of the contours are not displayed. 

I've attached a couple screenshots as an example. 
![selection_043](https://cloud.githubusercontent.com/assets/955351/13901010/1175f2ca-edeb-11e5-9f64-7361cf90bfc9.png)
![selection_042](https://cloud.githubusercontent.com/assets/955351/13901011/11824cd2-edeb-11e5-9d8a-8b7010862c02.png)

Let me know if you'd like any other information. 

Here's the python log from qgis: https://gist.github.com/skorasaurus/e806d576c9141eb677ad

**Comment by ccrook** [comment]
Hi

Once the contours are created they are just standard memory layers.  So the display of them is nothing to do with the plugin - that is just QGIS displaying a memory layer.  Which makes it odd that they don't display at some scales.  

It would be interesting to save them to another format, for example GML or shapefile, and then see whether the issue persists (right click on layer in layers dialog, then save as..)

Cheers
Chris

**Comment by skorasaurus** [comment]
Hi Chris, 

Thanks for the help. When I tried to save it as a layer as a SHP or GML, I receive this error: 

``Export to vector file failed.
Error: Feature write errors:
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Invalid variant type for field walking_time_min[1]: received PyQt_PyObject with type 
Only 0 of 6 features written.``

**Comment by ccrook** [comment]
That is odd indeed!  Are you able to create a test data set I could use to debug this?  And the settings you are using in the contour plugin.  That would be great - thanks

**Comment by skorasaurus** [comment]
Yes! 

I reproduced this again with my test data set which is a bbox extraction of my original data set. 
Data is available here as a [https://gist.github.com/a87ca618065643cd55a3afd60b81d2bb](gist). 

My settings for contour creation 

![selection_047](https://cloud.githubusercontent.com/assets/955351/14151470/0942460a-f67c-11e5-8932-cbc74013b1ad.png)

and my settings for saving as a vector layer 
![selection_046](https://cloud.githubusercontent.com/assets/955351/14151471/09472fb2-f67c-11e5-9457-23337526a803.png)

Let me know if there's anything else that I can do to help.

**Comment by ccrook** [comment]
That's great - I've been able to reproduce the saving issue (even though I'm currently using QGIS 2.8).  I'll get back to you on this shortly!

I'm not yet sure whether this relates to the original display issue.  One thing I noticed was a warning in the python log about:

DuplicatePointWarning: Input data contains duplicate x,y points; some values are ignored.

It could be worth ticking the "Apply point thinning" option in the contour dialog to see if it makes any difference.  I tried this and still get the issue with saving the layer.  As yet I've not seen the display issue so I don't know if this is affected or not.

**Comment by ccrook** [comment]
Hi

I've uploaded a new version, 1.3.6, which should fix the saving issue.  I don't think this will fix your original issue with displaying though.  So.. would be great if you can set up a test data set for that too!
(or if you have that issue with the current test data set, can you provide details as to how to view it, maybe a zipped up project and a saved bookmark for the offending view).

Thanks for providing the test data set for saving.

**Comment by skorasaurus** [comment]
Yes, the new version, 1.3.6, fixed the saving issue. In case you're still interested, the display issue is still occurring. 

To reproduce: 
* Open QGIS
* add this geojson file https://gist.github.com/skorasaurus/a87ca618065643cd55a3afd60b81d2bb as a vector layer. 
* run contour plugin 
![screenshot from 2016-04-18 18 38 24](https://cloud.githubusercontent.com/assets/955351/14623059/b9276be8-059c-11e6-8789-02447a733280.png)
Above is one example of the settings that I used; I've also ran it with point thinning of 0.00005 and had the same display issue.

**Comment by ccrook** [comment]
Yes interested - and have now been able to replicate.  I've just installed QGIS 2.14 and it is very apparent from in that.  However also showing in earlier versions of QGIS.  Exported the contour layer to GML and .CSV, and failed in both.  Also noticed that the objects that don't displayed in some views also not selectable, so clearly there is some error in the definition of their extents.  Will look in detail at the geometry to see whether the geometry definition is invalid (contour plugin error) or QGIS is handling it incorrectly. Great to have this example.  Thanks.

**Comment by ccrook** [comment]
Further ... have established that the geometry generated by matplotlib are not valid.  Will give some consideration to how to convert to valid geometry.  Using simple .buffer(0.0) did not work - resulting geometry was valid but did not include most of area supposed to be in contour.

**Comment by skorasaurus** [comment]
Yes, thank you for finding the geometry bug as well! I was going to make a separate report about that last night but feel asleep. 

Regarding the geometry bug, several of the polygons had its

**Comment by mwiesbau** [comment]
I have the same issue and I found a workaround by using the commands as outlined below.
From GIS StackExchange

[Vector > Geometry Tools > Multiparts to Singlepart...](http://gis.stackexchange.com/questions/131996/shapefile-layers-in-qgis-dont-display-unless-i-zoom-out)
